app/Providers/NetworkServiceProvider.php

```diff
@@ -24,4 +26,9 @@ public function register()
             );
         });
     }
+
+    public function provides()
```

Comment:
Not seen this before - nice!

Reply:
Deferred providers - means we can defer loading DI's that involve database queries (for example) until they're needed!